pyfolio/perf_attrib.py

```diff
@@ -0,0 +1,148 @@
+                factor_returns,
+                strategy_daily_returns,
+                strategy_daily_holdings):
+    """
+    Does performance attribution given risk info.
+
+    Parameters
+    ----------
+    factor_loadings : pd.DataFrame
+        Factor loadings for all days in the date range, with date and ticker as
+        index, and factors as columns.
+        - Example:
+                               momentum  reversal
+            dt         ticker
+            2017-01-01 AAPL   -1.592914  0.852830
+                       TLT     0.184864  0.895534
+                       XOM     0.993160  1.149353
+            2017-01-02 AAPL   -0.140009 -0.524952
+                       TLT    -1.066978  0.185435
+                       XOM    -1.798401  0.761549
+
+    factor_returns : pd.DataFrame
+        Returns by factor, with date as index and factors as columns
+        - Example:
+                        momentum  reversal
+            2017-01-01  0.002779 -0.005453
+            2017-01-02  0.001096  0.010290
+
+    strategy_daily_returns : pd.DataFrame
+        Returns for each day in the date range.
+        - Example:
+                            AAPL       TLT       XOM
+            2017-01-01 -0.014092 -0.003938 -0.001508
+            2017-01-02  0.011921  0.003039 -0.001546
+
+    strategy_daily_holdings: pd.Series
+        Daily holdings for all days in the date range, indexed by date
+        and ticker
+        - Example:
+            dt          ticker
+            2017-01-01  AAPL      71
+                        TLT       93
+                        XOM       10
+            2017-01-02  AAPL      71
+                        TLT       16
+                        XOM       71
+
+    Returns
+    -------
+    exposures : pd.DataFrame
+        df with factors as columns, and datetimes as index
+        - Example:
+                                 momentum    reversal
+            dt         ticker
+            2017-01-01 AAPL   -113.096901   60.550926
+                       TLT      17.192383   83.284658
+                       XOM       9.931600   11.493529
+            2017-01-02 AAPL     -9.940674  -37.271562
+                       TLT     -17.071643    2.966952
+                       XOM    -127.686503   54.070000
+
+    perf_attribution : pd.DataFrame
+        df with factors, common returns, and specific returns as columns,
+        and datetimes as index
+        - Example:
+                        momentum  reversal  common_returns  specific_returns
+            dt
+            2017-01-01  0.249087  0.935925        1.185012          1.185012
+            2017-01-02 -0.003194 -0.400786       -0.403980         -0.403980
+    """
+
+    risk_exposures = factor_loadings.multiply(strategy_daily_holdings,
+                                              axis='rows')
+    risk_exposures_portfolio = risk_exposures.groupby(level='dt').sum()
+    perf_attrib_style = risk_exposures_portfolio.multiply(factor_returns)
```

Comment:
`perf_attrib_style` will also do attribution to sectors, right? E.g. as far as pyfolio is concerned, the technology factor is no different from the momentum factor: its just another common factor. So maybe `perf_attrib_style` isn't the best name here?